scripts/us_census/decennial/process.py

```diff
@@ -0,0 +1,303 @@
+"""A script to clean Decennial Census Redistricting data from FTP site. """
+
+import csv
+import glob
+import io
+import os
+import zipfile
+
+from absl import app
+from absl import flags
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    'raw_data_path', 'scratch', 'Path to the equivalent of '
+    'https://www2.census.gov/programs-surveys/decennial/')
+flags.DEFINE_string('output_path', 'output', 'Output directory')
+flags.DEFINE_boolean('verbose', False, 'Print debug info')
+
+_YEARS = ['2010', '2020']
+_DIR_PREFIX = 'data/01-Redistricting_File--PL_94-171'
+
+#
+# Mappings for 2020 Geo File.
+#
+# - https://user-images.githubusercontent.com/4375037/130695847-b0955d19-9b8b-4021-b195-86357cfb03ec.png
+#
+
+# Summary level in geo file.
+_GEOF_2020_SUMLEV_COL = 2
+# Logical Record Number in geofile.
+_GEOF_2020_LOGRECNO_COL = 7
+
+# summary-level -> [column1, column2, ...] (from 0) which form parts of DCID
+#
+# Relevant summary levels: https://user-images.githubusercontent.com/4375037/130695894-365a96b6-8bec-4a98-ab91-fe90ea3c2e57.png
+_GEOF_2020_DCID_MAP = {
+    # State
+    '040': [12],
+    # County
+    '050': [12, 14],
+    # CouSub
+    '060': [12, 14, 17],
+    # Tract
+    '140': [12, 14, 32],
+    # BlockGroup
+    '150': [12, 14, 32, 33],
+    # Place
+    '160': [12, 29],
+}
+
+#
+# Mappings for 2010 Geo File. These are not really delimited. Rather, they have
+# start-position and length for the fields.
+#
+# - https://user-images.githubusercontent.com/4375037/130695985-6996cd98-66c9-4a59-87c7-48b9e389e2bc.png
+#
+_GEOF_2010_SUMLEV = (9, 3)
+_GEOF_2010_LOGRECNO = (19, 7)
+
+# summary-level -> [(start1, length1), (start2, length2), ...]
+#
+# Relevant summary levels: https://user-images.githubusercontent.com/4375037/130695894-365a96b6-8bec-4a98-ab91-fe90ea3c2e57.png
+_GEOF_2010_DCID_MAP = {
+    # State
+    '040': [(28, 2)],
+    # County
+    '050': [(28, 2), (30, 3)],
+    # CouSub
+    '060': [(28, 2), (30, 3), (37, 5)],
+    # Tract
+    '140': [(28, 2), (30, 3), (55, 6)],
+    # BlockGroup
+    '150': [(28, 2), (30, 3), (55, 6), (61, 1)],
+    # Place
+    '160': [(28, 2), (46, 5)],
+}
+
+# US Summary level.
+_US_SUMLEV = '010'
+
+# Logical Record Number in datafile is same across years.
+# 2020 - https://user-images.githubusercontent.com/4375037/130696087-14199c77-58f7-40cb-8010-3355356d7275.png
+# 2010 - https://user-images.githubusercontent.com/4375037/130696096-a40d7306-1707-4359-92af-a7bd391e2623.png
+_DATAF_LOGRECNO_COL = 4
+
+# Delimiter for data file varies across years.
+_DATAF_DELIM_CHAR = {
+    '2010': ',',
+    '2020': '|',
+}
+
+# Table name as it appears in Data Dictionary Reference:
+# https://user-images.githubusercontent.com/4375037/130696166-da194721-c8e1-484e-8722-09202d3fe5c3.png
```

Comment:
maybe also add the source for these images once in the file header? unfortunately, unlike screenshot tool there isn't a ready link generated.

Reply:
Added a comment noting the source docs.